### PR TITLE
Improve documentation of toric varieties

### PR DESCRIPTION
**Pull request by HereAround**
- Restructure content (in particular the constructors for toric varieties).
- Provide more information to filter the docstring, as currently documentation for many other methods is displayed.

### CODE REVIEW

**Comment by fingolfin** [comment]
This now has a conflict, could you please rebase this (and drop your workaround for the doctest failure in lieu of the fix now on master)?

File: src/ToricVarieties/NormalToricVarieties/constructors.jl [hunk 1]
```diff
@@ -119,6 +119,21 @@ end
 
 export NormalToricVariety
 
+
+@doc Markdown.doc"""
+    AffineNormalToricVariety(v::NormalToricVariety)
+
+For internal design, we make a strict distinction between normal toric varieties and affine toric varieties. Given a normal toric variety `v` which is affine, this method turns it into an affine toric variety.
```
Comment (fingolfin) [inline src/ToricVarieties/NormalToricVarieties/constructors.jl]: We don't have strict rules on this, but I think it's a good idea in general to wrap lines at somewhere between 80 - 100 columns. If only to make reviews in the GitHub web interface less of a chore (right now this line is only shown partially to me and I have to scroll horizontally a lot to read all of it)

File: src/ToricVarieties/ToricDivisors.jl [hunk 2]
```diff
@@ -71,7 +71,7 @@ export DivisorOfCharacter
 ######################
 
 @doc Markdown.doc"""
-    iscartier(d::ToricDivisor)
+    iscartier(td::ToricDivisor)
 
 Checks if the divisor `d` is Cartier.
```
Comment (fingolfin) [inline src/ToricVarieties/ToricDivisors.jl]:
```suggestion
Checks if the divisor `td` is Cartier.
```

**Review by fingolfin — COMMENTED** [review]
Looks fine to me -- but it is marked as draft... does that mean more is to come?!

File: src/ToricVarieties/NormalToricVarieties/constructors.jl (hunk 1 above)
Comment (HereAround) [reply src/ToricVarieties/NormalToricVarieties/constructors.jl]: I will adjust this.

Yes, I have been making a few more adjustments (due to the other doctest failures). It is almost ready, and I hope to push the final version in the next couple of minutes. Will then marked this as "ready for review".

File: src/ToricVarieties/ToricDivisors.jl (hunk 2 above)
Comment (HereAround) [reply src/ToricVarieties/ToricDivisors.jl]: Yes, this is also fixed in the version that I will force-push in a few minutes - hopefully. Thanks for pointing this out.

File: docs/src/ToricVarieties/NormalToricVarieties.md [hunk 3]
```diff
@@ -22,118 +22,125 @@ the affine and non-affine case:
...
+AffineNormalToricVariety(v::NormalToricVariety)
 ```
 
+### Cyclic Quotient Singularities
 
-## Properties of toric varieties
+Cyclic quotient singularities are quotients of $\mathbb{C}^2$ by the action of
+$\mathbb{Z}/n\mathbb{Z}$ acting via 
+$$\left(\begin{array}{cc}\xi & 0\\0 & \xi^q\end{array}\right)$$,
+where $\xi$ is a $n$-th root of unity, and $0<q<n$ are two coprime integers.
```
Comment (fingolfin) [inline docs/src/ToricVarieties/NormalToricVarieties.md]: For me, `$0<q<n$ are two coprime integers` is not clear (as `$0<q<n$` is/are not "two integers"). How about this:
```suggestion
where $\xi$ is a $n$-th root of unity, and $0<q<n$ is coprime with $n$.
```
or even better
```suggestion
where $\xi$ is a $n$-th root of unity, and $q$ is coprime with $n$.
```
(I don't think it is necessary to mention `0<q<n$ for the definition, is it?)

File: docs/src/ToricVarieties/NormalToricVarieties.md [hunk 4]
```diff
@@ -22,118 +22,125 @@ the affine and non-affine case:
...
-isnormal
-isorbifold
-isprojective
-isq_gorenstein
-```
+For the notation we rely on [Chr91](@cite) and [Ste91](@cite).
 
+!!! warning
+    Note that the notion of Hirzebruch-Jung continued fraction differs from the
+    commonly known continued fraction.
```
Comment (fingolfin) [inline docs/src/ToricVarieties/NormalToricVarieties.md]: This remarks is a bit "out of the blue" here. I assume it makes sense if one has read the two citations you give above? Then I'd mention this, e.g.
> CITE and CITE use Hirzebruch-Jung continued fractions, which differ from the usual  continued fractions found in most literature, and also used by the rest of OSCAR.

**Review by fingolfin — APPROVED** [review]

File: docs/src/ToricVarieties/NormalToricVarieties.md (hunk 3 above)
Comment (HereAround) [reply docs/src/ToricVarieties/NormalToricVarieties.md]: This is ok for me. @lkastner, I believe you wrote this code (while I merely copied it to a different location). Are you ok too? Are we overlooking something?

File: docs/src/ToricVarieties/NormalToricVarieties.md (hunk 4 above)
Comment (HereAround) [reply docs/src/ToricVarieties/NormalToricVarieties.md]: Ok for me, but again written by @lkastner (and I merely copied it to a different location). @lkastner are you ok with this change?

File: docs/src/ToricVarieties/NormalToricVarieties.md (hunk 3 above)
Comment (lkastner) [reply docs/src/ToricVarieties/NormalToricVarieties.md]: This is not ok, as this omits the other two conditions, namely $0<q$ and $q<n$. Is this a formulation issue? The formulation "are coprime integers" and "are two coprime integers" is not that uncommon, but if it makes you happy we can say "coprime with". So I then suggest:
```suggestion
where $\xi$ is a $n$-th root of unity, and $q$ and $n$ are integers, such that $q$ is coprime with $n$, and $0<q<n$.
```

File: docs/src/ToricVarieties/NormalToricVarieties.md (hunk 4 above)
Comment (lkastner) [reply docs/src/ToricVarieties/NormalToricVarieties.md]: Sounds good to me, so I make this concrete:
```suggestion
    Note that [Chr91](@cite) and [Ste91](@cite) use Hirzebruch-Jung continued fraction, which differ from the
    commonly known continued fraction from literature and used in the rest of OSCAR.
```

File: docs/src/ToricVarieties/NormalToricVarieties.md (hunk 3 above)
Comment (HereAround) [reply docs/src/ToricVarieties/NormalToricVarieties.md]: Thank you Lars.

File: docs/src/ToricVarieties/NormalToricVarieties.md (hunk 4 above)
Comment (HereAround) [reply docs/src/ToricVarieties/NormalToricVarieties.md]: Thank you Lars.

**Pull request merged**